src/store/modules/user/actions.ts

```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current eComStore information
    */
-  async setEcomStore({ commit, dispatch }, payload) {
+  async setEcomStore({ commit, dispatch, getters }, payload) {
     dispatch('job/clearJobState', null, { root: true });
-    commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.eComStore);
-    dispatch('getShopifyConfig', payload.eComStore.productStoreId);
+    let productStore;
+
+    if(payload.productStore) {
+      commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.productStore);
+    } else {
+      productStore = getters.getUserProfile.stores.find((store: any) => store.productStoreId === payload.productStoreId);
+    }
+    
+    dispatch('getShopifyConfig', productStore ? productStore.productStoreId : payload.productStore.productStoreId);
```

Comment:
Once we have `productStore` set for both the cases, we will not need this ternary operation we could use `productStore.productStoreId` for both the cases